covalent_dispatcher/_service/app_dask.py

```diff
@@ -0,0 +1,225 @@
+from __future__ import annotations
+
+import asyncio
+import os
+from logging import Logger
+from multiprocessing import Process, current_process
+from threading import Thread
+
+import dask.config
+from dask.distributed import Client, LocalCluster
+from distributed.core import Server, rpc
+from distributed.node import ServerNode
+
+from covalent._shared_files import logger
+from covalent._shared_files.config import get_config, set_config, update_config
+from covalent._shared_files.utils import get_random_available_port
+
+app_log = logger.app_log
+
+# Configure dask to not allow daemon workers
+dask.config.set({"distributed.worker.daemon": False})
+
+
+# Trying to separate out the admin work to a different thread
+# because Dask process was not starting on macos since the process are spawned by default instead
+# of fork
+class DaskAdminWorker(Thread):
+    """
+    Runs the service handlers for the Dask cluster in a separate thread to circumvent
+    any pickling issues rising from running service handlers in
```

Comment:
Can you somehow merge this comment into the docstring, and also edit it a little bit? I'm not sure what "Trying to separate out the admin work" means exactly, did you separate out the admin work? Also "running service handlers in", what are they running in?

Reply:
The service handlers are running in a separate thread `DaskAdminWorker` ( I am subclassing the Thread object provided in the threading library). Initially the service handlers where all part of the main DaskCluster process i.e all the async methods where part of that class. On linux (ubuntu/rhel/fedora) this was working just fine because on linux machines python uses the fork method to create new processes. On MacOS the default method of choice is `spawn` and keeping all the handlers in the same process was resulting in dask not starting and the service handlers not running. As per the documentation, the spawn method starts a fresh python interpreter process and the child processes inherit the bare minimum resources required to run the process object's run method, other stuff is not inherited. since the handlers are async methods, they couldn't be pickled, thus I was running into a lot of pickling issues on cluster start on Mac.

To address this in a way where covalent would run equally well on both Linux/Mac... I had to move the CLI service handlers into a separate thread that starts along with DaskCluster.

Reply:
happy to hear alternatives that might result in simpler implementation